### PR TITLE
ParticipantItem menu button more like Teams

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

- Change the look and feel of ParticipantItem to be more like Teams.
https://user-images.githubusercontent.com/79475487/119892940-9a4b0100-beef-11eb-9865-2c8041ac1c1a.mp4
- Also added back the text'(you)' in ParticipantItem
- Big clean up ParticipantList

# Why

There is an ask to make ParticipantItem menu button to be like Teams

# How Tested

Through local storybook and sample apps

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook/stories/ParticipantList/ParticipantList.stories.tsx [hunk 1]
```diff
@@ -116,7 +118,7 @@ export const ParticipantList: () => JSX.Element = () => {
   };
 
   return (
-    <Stack>
+    <Stack style={{ height: '2rem' }}>
```
Comment (JamesBurnside) [inline packages/storybook/stories/ParticipantList/ParticipantList.stories.tsx:121]: What is this height ensuring? Ideally are components themselves handle awkward edge cases or min-heights

File: packages/react-components/src/components/ParticipantList.tsx [hunk 2]
```diff
@@ -32,12 +25,13 @@ export type ParticipantListProps = {
   onParticipantRemove?: (userId: string) => void;
 };
 
-const getDefaultRenderer = (
+const defaultRenderer = (
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ParticipantList.tsx:28]: nit rename to include participants as default renderer doesn't indicate what it is rendering

```suggestion
const renderParticipants = (
```
or
```suggestion
const onRenderParticipantsDefault = (
```

File: packages/react-components/src/components/ParticipantItem.tsx [hunk 3]
```diff
@@ -85,26 +77,54 @@ export const ParticipantItem = (props: ParticipantItemProps): JSX.Element => {
...
+              title="More options"
+              className={menuButtonContainerStyle}
+              onClick={() => setMenuHidden(false)}
+            >
+              <MoreIcon outline={!menuButtonHovered} />
+            </div>
+          ) : (
+            onRenderIcon(props)
+          )}
+          {menuItems && (
```
Comment (JamesBurnside) [inline packages/react-components/src/components/ParticipantItem.tsx:116]: nit: should this check the menuItems.length as well here (i.e. don't show the contextual menu if there are no menu items inside the menu)
```suggestion
          {menuItems && menuItems.length > 0 && (
```

Comment (JamesBurnside) [reply packages/react-components/src/components/ParticipantItem.tsx:116]: (I guess this isn't necessary actually as the `menuHidden` will never be false when menuItems.length === 0)

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-components/src/components/ParticipantItem.tsx [hunk 4]
```diff
@@ -85,26 +77,54 @@ export const ParticipantItem = (props: ParticipantItemProps): JSX.Element => {
...
-          hidden={menuHidden}
-          target={clickEvent ?? containerRef}
-          onItemClick={hideMenu}
-          onDismiss={hideMenu}
-          directionalHint={DirectionalHint.bottomLeftEdge}
-          className={mergeStyles(styles?.menu)}
-        />
+        <Stack horizontal={true} className={mergeStyles(iconContainerStyle, styles?.iconContainer)}>
+          {menuItems && menuItems.length > 0 && (itemHovered || !menuHidden) ? (
+            <div
```
Comment (allenplusplus) [inline packages/react-components/src/components/ParticipantItem.tsx:104]: Is this div only the size of the icon? So you must click on the icon to show the menu? How does this work for mobile users, if the itemHovered if false, will this div still show up? Can the mobile users access the menu if they do not have the hover ability?

**Review by allenplusplus — APPROVED** [review]

**Pull request merged**